include/mbedtls/ssl.h

```diff
@@ -525,6 +526,160 @@ typedef void mbedtls_ssl_set_timer_t( void * ctx,
  */
 typedef int mbedtls_ssl_get_timer_t( void * ctx );
 
+#if defined(MBEDTLS_SSL_ASYNC_PRIVATE_C)
+#if defined(MBEDTLS_X509_CRT_PARSE_C)
+/**
+ * \brief           Callback type: start external signature operation
+ *
+ *                  Callback to start a signature operation using an
+ *                  external processor. The parameter \c cert contains
+ *                  the public key; it is up to the callback function to
+ *                  look up the associated private key or a handle to the
+ *                  private key.
+ *
+ *                  This function must start the signature operation.
+ *                  It is expected to be non-blocking, i.e. typically
```

Comment:
I think here and below, it would be good to explicitly mention the two cases: the callback blocks until it completed the operation, or it returns early with the necessary information to resume later. Both are equally fine (even though this PR is mainly motivated by the second case). I think mentioning the two cases explicitly helps the user make sense of all the conditional things below.

By the way, I feel like this feature would benefit from a KB article which introduces motivations and concepts in a user-friendly order that doxygen can't provide. Is anything planned on this front?

Reply:
I'm planning a KB article, I haven't written it yet. I do think it would be difficult to figure out this feature by reading function/type descriptions.